File: src/main/java/failchat/handlers/CommonHighlightHandler.java
package failchat.handlers;

import failchat.core.Message;
import failchat.core.MessageHandler;
import org.apache.commons.lang.StringUtils;

public class CommonHighlightHandler implements MessageHandler {
    private String appeal;

    public CommonHighlightHandler(String channel) {
        this.appeal = channel + ',';
    }

    @Override
    public void handleMessage(Message message) {
        if (StringUtils.containsIgnoreCase(message.getText(), appeal)) {
            message.setHighlighted(true);
        }
    }
}

File: src/main/kotlin/failchat/core/chat/handlers/CommonHighlightHandler.kt
package failchat.core.chat.handlers

import failchat.core.chat.ChatMessage
import failchat.core.chat.MessageHandler

class CommonHighlightHandler<in T>(username: String) : MessageHandler<T>
where T : ChatMessage {

    private val appeal: String = username + ','

    override fun handleMessage(message: T) {
        if (message.text.contains(appeal, ignoreCase = true)) {
            message.highlighted = true
        }
    }

}
